File: Java/E-montazysta/src/main/java/com/emontazysta/model/Order.java
package com.emontazysta.model;

import com.emontazysta.enums.TypeOfPriority;
import com.emontazysta.enums.TypeOfStatus;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import java.util.Date;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private TypeOfStatus typeOfStatus;

    private Date plannedStart;

    private Date plannedEnd;

    private Date createdAt;

    private Date editedAt;

    private TypeOfPriority typeOfPriority;
}

File: Android/app/src/main/java/com/example/e_montazysta/data/model/Order.kt
package com.example.e_montazysta.data.model

import com.example.e_montazysta.ui.location.LocationDAO
import com.example.e_montazysta.ui.order.OrderPriority
import com.example.e_montazysta.ui.order.OrderStatus
import java.util.Date

data class Order(
    val id: Int,
    val name: String,
    val priority: OrderPriority,
    val status: OrderStatus,
    val plannedStart: Date?,
    val plannedEnd: Date?,
    val client: Int?,
    val foreman: User?,
    val manager: User?,
    val specialist: User?,
    val salesRepresentative: User?,
    val location: LocationDAO?,
    val orderStages: List<Int>,
    val createdAt: Date?,
    val editedAt: Date?
)
